### ISSUE TITLE
Migrate Google Analytics to Keen

### ISSUE THREAD

**Issue by shakeelmohamed** [zen-audio-player/zen-audio-player.github.io#26]
## Priority level 1
- [x] Create keen.io account
- [x] Have keen.io log as much data as GA has currently
- [x] Follow these guides: [media analytics](https://keen.io/guides/media-analytics/), [data enrichment](https://keen.io/docs/api/?javascript#data-enrichment), [data collection](https://keen.io/docs/data-collection/), [data analysis](https://keen.io/docs/data-analysis/)
- [x] Decide between [keen-js](https://github.com/keen/keen-js) & [keen-tracking](https://github.com/keen/keen-tracking.js) - looks like tracking is not ready for GA, keen-js it is!
- [x] Get Keen to log GA-like demographics data
- [ ] Add an event for when the tweet button is clicked to log share stats
- [ ] Remove GA
- [ ] Find a way to export GA to keen.io (maybe follow [Moz's guide](https://moz.com/blog/how-to-export-google-analytics-data-to-excel-via-the-api))
## Priority level 2
- [X] Create a new repo for Keen dashboards with a `gh-pages` branch: https://github.com/zen-audio-player/stats
- [x] [Example dashboard](http://keen.github.io/dashboards/examples/connected-devices/) more [here](http://keen.github.io/dashboards/examples/)
- [x] [data viz](https://keen.io/docs/data-visualization/) tutorial
- [ ] Start building keen.io dashboards - see current progress [here](http://zen-audio-player.github.io/dashboards/examples/connected-devices/)
- [x] Add table for showing popular songs/artists by parsing video titles
- [ ] Add table for showing common errors
- [ ] chart for search vs. video ID vs. URL
- [ ] Video duration chart

I've built a really nice GA dashboard, it would be really cool to show these stats publicly via [keen.io](https://keen.io/).

![capture](https://cloud.githubusercontent.com/assets/1480063/8124646/b341f068-1092-11e5-99ca-70fc08f6e543.PNG)

**Comment by shakeelmohamed** [comment]
This could be helpful: https://github.com/onefrankguy/frankmitchell.org/blob/master/drafts/replacing-google-analytics-with-keen-io.md

**Comment by shakeelmohamed** [comment]
Made the repo!

**Comment by shakeelmohamed** [comment]
Made tons of progress on the feature/keen.io branch, sent their team a follow-up email for next steps :clock10:

**Comment by shakeelmohamed** [comment]
The data we get into Keen for playing a video is roughly like so

```json
{
    "parsed_user_agent": {
        "device": {
            "family": "Other"
        },
        "os": {
            "major": "10",
            "patch_minor": null,
            "minor": "9",
            "family": "Mac OS X",
            "patch": "5"
        },
        "browser": {
            "major": "46",
            "minor": "0",
            "family": "Chrome",
            "patch": "2490"
        }
    },
    "author": "",
    "seconds": 347,
    "title": "The Glitch Mob -  Starve the Ego, Feed the Soul",
    "parsed_page_url": {
        "query_string": {
            "v": "koJv-j1usoI"
        },
        "path": "/redacted/zen-audio-player.github.io/index.html",
        "domain": "",
        "protocol": "file",
        "anchor": ""
    },
    "keen": {
        "timestamp": "2015-11-25T09:29:17.039Z",
        "created_at": "2015-11-25T09:29:17.039Z",
        "id": "56557f6dc2266c1b5a1fc951"
    },
    "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.80 Safari/537.36",
    "ip_geo_info": {
        "province": null,
        "city": null,
        "postal_code": null,
        "continent": "North America",
        "country": "United States"
    },
    "ip_address": "xxx.xxx.xxx.xxx",
    "page_url": "file:///redacted/zen-audio-player.github.io/index.html?v=koJv-j1usoI#"
}    
```

**Comment by shakeelmohamed** [comment]
I have been trying to setup a call with the Keen folks to help move this along. This is still blocked by them unless we get (or become) a Keen expert :hourglass:

**Comment by shakeelmohamed** [comment]
The table viz is using [Google's table viz](https://developers.google.com/chart/interactive/docs/gallery/table#example) under the hood which has sorting, etc

**Comment by shakeelmohamed** [comment]
Keen has changed business models and have removed thr free open source plan we were using before. I’ll open another issue to remove all keen code 😤
